tests/securedrop/install.pm

```diff
@@ -0,0 +1,135 @@
+
+    # Enable "presentation mode" to prevent the screen from going dark
+    assert_and_click('disable-screen-blanking-click-power-tray-icon');
+    assert_and_click('disable-screen-blanking-click-presentation-mode');
+    send_key('esc');
+
+
+    x11_start_program('xterm');
+    send_key('alt-f10');  # maximize xterm to ease troubleshooting
+
+    curl_via_netvm;
+
+    assert_script_run('set -o pipefail'); # Ensure pipes fail
+
+    # NOTE: These are done via qvm-run instead of gnome-terminal so that we
+    # can know in case they failed.
+    assert_script_run('qvm-run -p work -- gpg --keyserver hkps://keys.openpgp.org --recv-key "2359 E653 8C06 13E6 5295 5E6C 188E DD3B 7B22 E6A3"');
+    assert_script_run('qvm-run -p work -- "gpg --armor --export 2359E6538C0613E652955E6C188EDD3B7B22E6A3 > securedrop-release-key.pub"');
+    assert_script_run('qvm-run -p work -- sudo rpmkeys --import securedrop-release-key.pub');
+    assert_script_run('qvm-run -p work -- "echo -e \"[sd]\nenabled=1\nbaseurl=https://yum.securedrop.org/workstation/dom0/f37\nname=boostrap\"  | sudo tee /etc/yum.repos.d/securedrop-temp.repo"');
+    assert_script_run('qvm-run -p work -- dnf download -y securedrop-workstation-dom0-config');
+    assert_script_run('qvm-run -p work -- "rpm -Kv securedrop-workstation-dom0-config-*.rpm"');  # TODO confirm output is correct
+    assert_script_run('qvm-run -p work -- "cat /home/user/securedrop-workstation-dom0-config-*.rpm" > securedrop-workstation.rpm');
+    assert_script_run('sudo dnf -y install securedrop-workstation.rpm');
+    assert_script_run('echo {\"submission_key_fpr\": \"65A1B5FF195B56353CC63DFFCC40EF1228271441\", \"hidserv\": {\"hostname\": \"bnbo6ryxq24fz27chs5fidscyqhw2hlyweelg4nmvq76tpxvofpyn4qd.onion\", \"key\": \"FDF476DUDSB5M27BIGEVIFCFGHQJ46XS3STAP7VG6Z2OWXLHWZPA\"}, \"environment\": \"prod\", \"vmsizes\": {\"sd_app\": 10, \"sd_log\": 5}} | sudo tee /usr/share/securedrop-workstation-dom0-config/config.json');
+    assert_script_run('curl https://raw.githubusercontent.com/freedomofpress/securedrop/d91dc67/securedrop/tests/files/test_journalist_key.sec.no_passphrase | sudo tee /usr/share/securedrop-workstation-dom0-config/sd-journalist.sec');
+    assert_script_run('sdw-admin --validate');
+
+    assert_script_run('env xset -dpms; env xset s off', valid => 0, timeout => 10); # disable screen blanking during long command
+    assert_script_run('sdw-admin --apply | tee /tmp/sdw-admin-apply.log',  timeout => 2400);  # long timeout due to slow virt.
+    upload_logs('/tmp/sdw-admin-apply.log');
+
+    # Reboot system
+    script_run('sudo reboot', timeout => 0);
+    # assert_screen ["bootloader", "luks-prompt", "login-prompt-user-selected"], 300;
+    $self->handle_system_startup;
+
+    # Accept Whonix systemcheck prompts (for sys-whonix and sd-whonix)
+    while (check_screen('whonix-systemcheck-derived-repo-accept', timeout => 300)) {
+        assert_and_click('whonix-systemcheck-derived-repo-accept');
+    }
```

Comment:
I'd recommend adding `wait_still_screen` in such a loop. Or just `sleep` after clicking. Otherwise it may loop quickly while the window is still visible.

And also, add some timeout (max number of iterations), otherwise if it will keep looping, you'll eventually hit hard timeout and that kills the job and skips collecting logs on failure.

Reply:
You should match just the information you are interested in. If you want to verify specific parts of this message, select just those messages etc. If the test really should fail if that window is a different size, then sure, you can select the whole window, but I doubt that's your goal...